packages/ui-web/src/slate/plugins/horizontal-divider.js

```diff
@@ -1,95 +1,92 @@
-import isHotkey from "is-hotkey";
-import { Editor, Node, Transforms } from "slate";
+import { Node, Range, Path } from "slate";
+import { useSelected, useFocused } from "slate-react";
 import { function as functionUtils } from "@shades/common/utils";
-import {
-  withBlockPrefixShortcut,
-  withEmptyBlockBackwardDeleteTransform,
-} from "../utils.js";
+import { withBlockPrefixShortcut } from "../utils.js";
 
 const { compose } = functionUtils;
 
 const ELEMENT_TYPE = "horizontal-divider";
 
-const findAbove = (editor, at = editor.selection) =>
-  Editor.above(editor, {
-    at,
-    match: (n) => n.type === ELEMENT_TYPE,
-  });
-
 const middleware = (editor) => {
-  const { isVoid, normalizeNode } = editor;
+  const { isVoid } = editor;
 
   editor.isVoid = (element) => {
     return element.type === ELEMENT_TYPE || isVoid(element);
   };
 
-  editor.normalizeNode = ([node, path]) => {
-    if (node.type !== ELEMENT_TYPE) {
-      normalizeNode([node, path]);
-      return;
-    }
-
-    const nodeString = Node.string(node);
-
-    // If the node is not empty, move text to new paragraph and insert divider above
-    if (nodeString.trim() !== "") {
-      const nodeStartPoint = Editor.start(editor, path);
-
-      Transforms.delete(editor, {
-        at: {
-          anchor: nodeStartPoint,
-          focus:
-            Editor.after(editor, nodeStartPoint) || Editor.end(editor, path),
+  return compose((editor) =>
+    withBlockPrefixShortcut(
+      {
+        prefix: "---",
+        elementType: ELEMENT_TYPE,
+        transform: ({ node, path }) => {
```

Comment:
uhh didn't know about this. I also thought using the normalize to add/remove nodes at insertion was a bit of a hack, but couldn't find a simple "when inserting this element, also run this code"

Reply:
Yeah I added the `transform` option for this case, just pushed it to main instead of this branch. My initial thought was that the block prefix shortcut was only going to be used to change the element type you’re currently in, but I realized some block types need more flexibiliy.